src/lib.rs

```diff
@@ -102,6 +103,10 @@ pub mod units {
     pub use crate::frequency::units::*;
 }
 
+/// Time-related error
+#[derive(Debug, Eq, PartialEq)]
+pub struct Error;
```

Comment:
Why is the error type a `struct` and not an `enum` as usual where variants can be added?

Reply:
I started with an enum, but basically just couldn't think of any variants to add at this point. There is currently only one failure mode, not being able to read the current instant. Any suggestions you have, would be much appreciated. Perhaps it's worth adding that as a variant is a good starting point? And adding the non_exhaustive attrib?

Reply:
I think I was not thinking correctly about the error types. I need to provide variants that the implementer will use to attempt to describe the reason the call to `now()` failed.